api/src/main/scala/Geoprocessing.scala

```diff
@@ -18,10 +18,122 @@ trait Geoprocessing extends Utils {
     * @param   input  The InputData
     * @return         A histogram of results
     */
-  def getRasterGroupedCount(input: InputData): Result = {
+  def getRasterGroupedCount(input: InputData): ResultInt = {
     val aoi = createAOIFromInput(input)
     val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
-    Result(rasterGroupedCount(rasterLayers, aoi))
+    ResultInt(rasterGroupedCount(rasterLayers, aoi))
+  }
+
+  /**
+    * For an InputData object, return a histogram of raster grouped average
+    * results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results
+    */
+  def getRasterGroupedAverage(input: InputData): ResultDouble = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
+    val targetRaster = input.targetRaster
+    val targetLayer = targetRaster match {
+      case Some(targetRaster) =>
+        cropSingleRasterToAOI(targetRaster, input.zoom, aoi)
+      case None =>
+        throw new Exception("Request data missing required 'targetRaster'.")
+    }
+
+    val average =
+      if (rasterLayers.isEmpty) rasterAverage(targetLayer, aoi)
+      else rasterGroupedAverage(rasterLayers, targetLayer, aoi)
+
+    ResultDouble(average)
+  }
+
+  /**
+    * Return the average pixel value from a target raster and a MultiPolygon
+    * area of interest.
+    *
+    * @param   targetRaster  The target raster
+    * @param   multiPolygon  The AOI as a MultiPolygon
+    * @return                A one element map averaging the pixel values
+    */
+  private def rasterAverage(
+    targetRaster: TileLayerCollection[SpatialKey],
```

Comment:
Consider calling this `targetLayer` instead of `targetRaster` to match the conventions used elsewhere.